DOCUMENTATION.md

```diff
@@ -51,8 +51,27 @@ Various Types used in this package can also be imported:
 import * as Types from '@oneblink/sdk/types'
 ```
 
-## OneBlink Role Permissions
+## Permissions
 
-Certain actions in the OneBlink SDK require that your key has particular permissions.
-These permissions can be set in your tenant's console. If you do not have access to your tenant's console, you will need to work with your Administrator to associate the permissions that you require with your key.
-Without the associated permissions for a particular action, you will be prohibited from completing that action using the OneBlink SDK. The associated permissions for each function are detailed on the function definitions.
+The majority of actions in this SDK require that your developer key has particular permissions. These permissions can be set in your tenant's console. If you do not have access to your tenant's console, you will need to work with your Administrator to associate the permissions that you require with your key. The associated permissions for each function are detailed on the function definitions.
+
+### Developer Key & Role
+
+Actions that require a developer key with a minimum role permission will have the following in the function definition:
+
+**Minimum Role Permission**
+
+The developer key used must be assigned a role with at least the permission documented for each action. E.g. If action's minimum role permission is _Forms: Read Only_, the role assigned to the developer key could have _Forms: Read Only_ or _Forms: Manager_.
+
+### Developer Key & App Association
+
+Actions that require a developer key to be assigned to an app will have the following in the function definition:
+
+**App Association Required**
+
+These actions require the developer key to be associated with an app that is associated with the form that is being actioned. A form can be associated with an app in the following ways:
```

Comment:
```suggestion
In the case of any actions relating to forms, the assigned app must be associated with the form that is being actioned. A form can be associated with an app in the following ways:
```

Reply:
This implies that not all of these actions relate to forms which is incorrect. They are all associated with a single form.

Reply:
Yeh but that might not always be the case. And tying the two concepts together is more for the user to understand IMO.

Reply:
I'm looking at it this way: Currently all the actions that require an app association just happen to be single form actions.